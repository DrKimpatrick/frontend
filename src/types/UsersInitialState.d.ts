import { User } from 'redux/action-types/user';

interface UsersInitialState {
  currentUser: {
    isLoggedIn: boolean;
    data: object | null | any;
    loading: boolean;
    error: null | object | any;
  };
  user?: User;
}

export default UsersInitialState;
