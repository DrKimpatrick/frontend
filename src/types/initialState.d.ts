import UsersInitialState from './UsersInitialState';

interface InitialStateInterface {
  users: UsersInitialState;
}

export default InitialStateInterface;
