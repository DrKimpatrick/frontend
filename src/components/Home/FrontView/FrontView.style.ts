import styled from 'styled-components';
import frontImage from 'assets/images/front.png';

const frontStyles = {
  FrontContainer: styled.div`
    background-image: url(${frontImage});
    background-repeat: no-repeat, repeat;
    background-size: cover;

    @media screen and (max-width: 480px) {
      display: flex;
      justify-content: center;
    }
  `,

  FrontText: styled.div`
    width: 40%;
    margin-left: 4rem;

    @media screen and (max-width: 480px) {
      width: 93% !important;
      margin-left: 0 !important;
    }

    @media (min-width: 480px) and (max-width: 768px) {
      width: 84% !important;
    }

    @media (min-width: 768px) and (max-width: 1200px) {
      width: 65% !important;
    }
  `,
  TopHeading: styled.h1`
    @media screen and (max-width: 768px) {
      margin-left: 0px;
    }
  `,
  FrontViewList: styled.ul`
    list-style: decimal;
    margin-left: 20px;

    @media screen and (max-width: 768px) {
      margin-left: 20px;
    }
  `,
  H5Title: styled.h5`
    @media screen and (max-width: 768px) {
      margin-left: 0px;
    }
  `,
  ButtonDiv: styled.div`
    @media screen and (max-width: 768px) {
      display: flex;
      justify-content: start;
      width: 100%;
    }
  `
};

export default frontStyles;
