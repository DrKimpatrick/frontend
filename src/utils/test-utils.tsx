import React, { FC, ReactElement } from 'react';
import { render } from '@testing-library/react';
import { SWRConfig } from 'swr';

const Wrapper: FC = ({ children }) => {
  return <SWRConfig value={{ dedupingInterval: 0 }}>{children}</SWRConfig>;
};

const customRender = (ui: ReactElement, options: any) =>
  render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
