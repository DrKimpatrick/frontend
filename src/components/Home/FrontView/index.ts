import FrontView from './FrontView';

export { FrontView };
