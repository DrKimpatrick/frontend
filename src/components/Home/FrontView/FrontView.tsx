import React, { FC } from 'react';
import { frontData } from 'utils/staticData';
import frontStyles from './FrontView.style';

const FrontView: FC = () => {
  return (
    <frontStyles.FrontContainer data-testid="frontView">
      <frontStyles.FrontText>
        <frontStyles.TopHeading className="pt-24 text-4xl text-white font-semibold">
          {frontData.title}
        </frontStyles.TopHeading>
        <div className="mt-4 text-white w-2/4 front-mob">
          <frontStyles.FrontViewList>
            {frontData.list1.map((item, index) => (
              <li key={index}>{item}</li>
            ))}
          </frontStyles.FrontViewList>
        </div>
        <div className="mt-4 text-white w-4/6 front-mob">
          <frontStyles.H5Title className="mb-1">
            {frontData.title2}
          </frontStyles.H5Title>
          <frontStyles.FrontViewList>
            {frontData.list2.map((item, index) => (
              <li key={index}>{item}</li>
            ))}
          </frontStyles.FrontViewList>
        </div>
        <frontStyles.ButtonDiv className="flex justify-between w-2/4 mt-8 pb-8 front-mob">
          <a
            href="/register"
            className="bg-green-500 w-40 text-white font-medium py-2 px-4 rounded text-center m-2"
          >
            Get Started
          </a>
          <button
            className="bg-white w-40 text-green font-medium py-2 px-4 rounded text-center m-2"
            style={{ color: 'green' }}
            type="button"
          >
            Watch Demo
          </button>
        </frontStyles.ButtonDiv>
      </frontStyles.FrontText>
    </frontStyles.FrontContainer>
  );
};

export default FrontView;
