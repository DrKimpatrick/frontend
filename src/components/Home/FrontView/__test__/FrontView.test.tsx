import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReactDom from 'react-dom';
import FrontView from '../FrontView';
import { isTSAnyKeyword } from '@babel/types';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';

let container: any;
let frontViewProps: any;

describe('Front View Component', () => {
  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);

    frontViewProps = {
      title: 'miss',
      description: 'Test description'
    };
  });

  afterEach(() => {
    cleanup;
    document.body.removeChild(container);
    container = null;
  });

  it('renders without crashing', () => {
    act(() => {
      ReactDom.render(<FrontView {...frontViewProps} />, container);
    });
  });

  it('should render title and desctiption correctly component correctly', () => {
    const { getByText } = render(
      <FrontView title="Title" description="This is the desctiption" />
    );
    expect(getByText('Title')).toBeInTheDocument();
    expect(getByText('This is the desctiption')).toBeInTheDocument();
  });

  it('should render Get Started and Watch Demo buttons correctly', () => {
    const { getByTestId } = render(
      <FrontView title="Title" description="This is the desctiption" />
    );

    expect(getByTestId('frontView')).toHaveTextContent('Get Started');
    expect(getByTestId('frontView')).toHaveTextContent('Watch Demo');
  });
});
