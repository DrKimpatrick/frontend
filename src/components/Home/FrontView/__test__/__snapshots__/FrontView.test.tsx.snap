// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Front View Component should render title and description correctly component correctly 1`] = `
<div
  className="sc-bdnylx jEnVUK"
  data-testid="frontView"
>
  <div
    className="sc-gtssRu gKWQKU"
  >
    <h1
      className="sc-dlnjPT hOHxbY pt-24 text-4xl text-white font-semibold"
    >
      Tech Talent QA
    </h1>
    <div
      className="mt-4 text-white w-2/4 front-mob"
    >
      <ul
        className="sc-hKFyIo bGDfPj"
      >
        <li>
          Certified education and employment
        </li>
        <li>
          Verified technical ability
        </li>
        <li>
          Continuous skills monitoring and updates
        </li>
      </ul>
    </div>
    <div
      className="mt-4 text-white w-4/6 front-mob"
    >
      <h5
        className="sc-eCApGN bmQaRK mb-1"
      >
        Tech Talent QA ensures
      </h5>
      <ul
        className="sc-hKFyIo bGDfPj"
      >
        <li>
          Members are current on the latest technology changes
        </li>
        <li>
          Companies are equipped with quality developers
        </li>
        <li>
          HR save time and money on recruiting potential candidates
        </li>
      </ul>
    </div>
    <div
      className="sc-jSFkmK fahdgH flex justify-between w-2/4 mt-8 pb-8 front-mob"
    >
      <a
        className="bg-green-500 w-40 text-white font-medium py-2 px-4 rounded text-center m-2"
        href="/register"
      >
        Get Started
      </a>
      <button
        className="bg-white w-40 text-green font-medium py-2 px-4 rounded text-center m-2"
        style={
          Object {
            "color": "green",
          }
        }
        type="button"
      >
        Watch Demo
      </button>
    </div>
  </div>
</div>
`;
